Add client-side impls of custom handshake timeout and custom ping-pong for event handler

platfowner opened the pull request:
Change summary:
- Implement custom handshake timeout for event handler
- Implement custom ping-pong for heartbeat 

Related issues:
- https://github.com/ainblockchain/ain-js/issues/171
- https://github.com/ainblockchain/ain-blockchain/issues/1260

jiyoung-an commented on src/event-manager/event-channel-client.ts:
```diff
@@ -187,32 +221,33 @@ export default class EventChannelClient {
+    this.sendPong();
+    // Heartbeat timeout
+    if (this._heartbeatTimeout) {
+      clearTimeout(this._heartbeatTimeout);
```
should the following code be included?
```
clearTimeout(this._heartbeatTimeout);
this._heartbeatTimeout = null;
```

platfowner replied:
Just done, thanks!

platfowner commented:
> LGTM, thank you for quickly fixing this issue 🙂 Could the test failure be because the blockchain isn't deployed yet?

Yes, it needs release of the server-side change: https://github.com/ainblockchain/ain-blockchain/pull/1261

jiyoung-an commented:
@platfowner I encountered the following error when building locally.
![스크린샷 2024-04-16 오전 1 45 05](https://github.com/ainblockchain/ain-js/assets/122012800/83033926-8573-4a66-9ceb-c0eca2303365)
I saw this similar [error](https://github.com/web3/web3.js/issues/6303), adjusted the version of @types/ws to 8.5.3 resolved the issue. 
Could you please downgrade the version to 8.5.3?

platfowner commented:
> @platfowner I encountered the following error when building locally. ![스크린샷 2024-04-16 오전 1 45 05](https://private-user-images.githubusercontent.com/122012800/322565457-83033926-8573-4a66-9ceb-c0eca2303365.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyMzM1MjAsIm5iZiI6MTcxMzIzMzIyMCwicGF0aCI6Ii8xMjIwMTI4MDAvMzIyNTY1NDU3LTgzMDMzOTI2LTg1NzMtNGE2Ni05Y2ViLWMwZWNhMjMwMzM2NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNlQwMjA3MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iY2I0NTllOTJjNTA1ZDBiMjM1YmIyOTg3NjM4NTUxMGFkYjBhNTIwNzA2MGRjODU5MDQ0YTg0NGZiNmRjMzQ3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.q7seWVwXUCm6W81lrD2xejDIXGlXJzrzmIOZzyOIeEE) I saw this similar [error](https://github.com/web3/web3.js/issues/6303), and adjusted the version of @types/ws to 8.5.3 resolved the issue. Could you please downgrade the version to 8.5.3?

just fixed, thanks!

platfowner commented:
Thanks for the review! @jiyoung-an Now merging.. and any issues reported would be handled by separate PRs.
